### ISSUE TITLE
Empty metadata scraped from TMDB

### ISSUE THREAD

**Issue by minikN** [tinyMediaManager/tinyMediaManager#417]
__What TMM version are you using?__
2.9.13

__release, pre-release, nightly, or directly from GitHub/branch?__
[AUR Release](https://aur.archlinux.org/packages/tiny-media-manager)

__What is the actual behaviour?__
Under special circumstances the TMDB scraper gets empty (no title, plot) metadata from TMDB.
See additional for more info

__What is the expected behaviour?__
It should scrape the episode correctly.

__Additional__
Have you attached the Logfile from the day it happened? [Yes](https://pastebin.com/qttXsgS4)
Arch Linux

I've investigated this a bit, and I think it's not an error with TMM, but with the TMDB API. This happens to me when I try to scrape the metadata for the S00 (specials) episodes of the tv show 'The X-Files'. Before I go on, I have named them correctly `S00/The X-Files - S00E01.avi`

This is how it looks after scraping the metadata from TMDB.
![auswahl_001](https://user-images.githubusercontent.com/1888338/48022161-b7a1ff00-e13a-11e8-86a4-3648a0d63789.png)

As you can see, it's in German. I've set my primary language to German and the Fallback language to english. I think what's happening here is related to the TMDB API. You see, there are no German titles and plots for the S0 episodes on TMDB, if you [visit their site](https://www.themoviedb.org/tv/4087-the-x-files/season/0?language=de-DE), you'll see the english versions.

However it doesn't work for their API. This is the response I get when trying to get the S00E01 via their API:
https://api.themoviedb.org/3/tv/4087/season/0/episode/1?api_key=1234&language=en-US

    {
      "air_date": "1998-06-19",
      "crew": [
        {
          "id": 12627,
          "credit_id": "52575c5a760ee36aaa2bccea",
          "name": "Frank Spotnitz",
          "department": "Writing",
          "job": "Writer",
          "gender": 2,
          "profile_path": "/4Yxj5h0IF6rta5OivhNwh4uSFBU.jpg"
        },
        {
          "id": 12524,
          "credit_id": "52575b6f760ee36aaa2b6fc0",
          "name": "Chris Carter",
          "department": "Writing",
          "job": "Writer",
          "gender": 2,
          "profile_path": "/cQl1drdtGH9CdUNKD0E1jc1q1Ww.jpg"
        },
        {
          "id": 12523,
          "credit_id": "52575b99760ee36aaa2b8031",
          "name": "Rob Bowman",
          "department": "Directing",
          "job": "Director",
          "gender": 2,
          "profile_path": "/5iUOH0n8ecLgzkr2O3aEWDknyeJ.jpg"
        }
      ],
      "episode_number": 1,
      "guest_stars": [],
      "name": "The X-Files: Fight the Future",
      "overview": "Blackwood, Texas: After falling through a hole in the ground, a young boy is infected by the alien Black Oil. One week later, in Dallas, a federal building is bombed by terrorists, killing five people, including the Blackwood boy. Mulder and Scully, now part of an FBI anti-terrorism unit, learn that the boy as well as three other victims were dead even before the bombing, cause of death: an alien virus. In the Texas desert, the agents find an agricultural complex that may hold the key to unlocking the government conspiracy surrounding the extraterrestrials and their lethal virus.",
      "id": 287570,
      "production_code": null,
      "season_number": 0,
      "still_path": "/nb2BkV8HIFuWmORvaIY7NsSRzVi.jpg",
      "vote_average": 9,
      "vote_count": 1
    }

All good here. However if I change the language to `de-DE`:

    {
      "air_date": "1998-06-19",
      "crew": [
        {
          "id": 12627,
          "credit_id": "52575c5a760ee36aaa2bccea",
          "name": "Frank Spotnitz",
          "department": "Writing",
          "job": "Writer",
          "gender": 2,
          "profile_path": "/4Yxj5h0IF6rta5OivhNwh4uSFBU.jpg"
        },
        {
          "id": 12524,
          "credit_id": "52575b6f760ee36aaa2b6fc0",
          "name": "Chris Carter",
          "department": "Writing",
          "job": "Writer",
          "gender": 2,
          "profile_path": "/cQl1drdtGH9CdUNKD0E1jc1q1Ww.jpg"
        },
        {
          "id": 12523,
          "credit_id": "52575b99760ee36aaa2b8031",
          "name": "Rob Bowman",
          "department": "Directing",
          "job": "Director",
          "gender": 2,
          "profile_path": "/5iUOH0n8ecLgzkr2O3aEWDknyeJ.jpg"
        }
      ],
      "episode_number": 1,
      "guest_stars": [],
      "name": "Folge 1",
      "overview": "",
      "id": 287570,
      "production_code": null,
      "season_number": 0,
      "still_path": "/nb2BkV8HIFuWmORvaIY7NsSRzVi.jpg",
      "vote_average": 9,
      "vote_count": 1
    }

Take a look at the `name` and `overview` keys. I don't know why they do it like that. But maybe there is something you guys can do? One option could be to fall back to english if the `overview` value is empty? Or simply add an option to scrape episodes (or better seasons) in a different language all together?

I don't know. Cheers.

**Comment by mlaggner** [comment]
has been fixed with https://github.com/tinyMediaManager/scraper-tmdb/commit/7fee0b51e7ec0cbcedf95490143f84ce04d4516b

**Comment by minikN** [comment]
Excuse my ignorance, but I can't get it to work. I cloned the repo amnd build the package. But when I right click on S0 and hit `scrape episode metadata` nothing happens. The status bar in the lower right corner appears for a fraction of a second and thats it.

[LOG](https://pastebin.com/T1AnvtmV)

**Comment by myron0815** [comment]
Yes, scraping this on season level does not work.
The fix was just for a fallback language, but not for this issue...

**Comment by mlaggner** [comment]
just had a look at the logs and saw that we did not include the patched tmdb scraper into the tmm build yet - that could be the cause for your issues.
in the next nightly build you should be able to scrape again

**Comment by minikN** [comment]
Thanks. I will wait for the next nightly and try again.

Allow me to mention another issue here I've been having (If you want me to create an new issue I can).

Is it by design that a freshly installed TMM will override existing meta data in nfo files? I've had the issue a couple of times that I've reinstalled TMM (due to this issue here), and whenever I add the file paths to my movies and tv shows (of which I scraped all the season posters ect before) it will go through every show again and download random artwork and overwrite what I have specified before. This this intended?

**Comment by minikN** [comment]
I've downloaded the latest 2.x NIGHTLY, which includes TMDB scraper `1.12-SNAPSHOT`. After that I added my TV Shows again, right clicked on Season 0 of The X-Files and selected `scrape episode metadata`. Nothing happens. The progress bar pops up for a fraction and disappears again. Here is the [log](https://pastebin.com/FMRxPU4A)

**Comment by mlaggner** [comment]
well where should be pull the metadata if it is not available?`
you screen shows Akte X - Season 0 - Episode 7

look at
https://www.themoviedb.org/tv/4087-the-x-files/season/0/episode/7?language=de-DE
and
https://www.themoviedb.org/tv/4087-the-x-files/season/0/episode/7?language=en-US

there is simply nothing here to fetch..

and according to your logs: looks like your TV show does not have a valid TMDB Id (it has a IMDB Id and TVDB Id) - although scraping TMDB with a TVDB Id _should_ work, your setup is a bit strange..
Did you scrape the TV show itself once with TMDB that it could fetch all needed IDs?

**Comment by minikN** [comment]
Hello.
I'm sorry, you're right about the covers / plots. It did fetch all it could. However, it still fetches the German titles, which aren't correct. I guess this is a problem with TMDB itself?

In terms of the ID, I can't really tell you. As I said, what I did was to manually remote all files except the videos files (for The X-Files). Then I added them back to TMM, right clicked on the TV-Show node and chose the first option `Suchen & Scrapen aller ausgewählten TV-Serien`. A dialog window popped up where it search TMDB and I chose the correct entry for The X-Files. That's all I did.
